listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -695,6 +695,32 @@ def delete(self, musicbrainz_id):
             self.log.error("Cannot delete listens for user: %s" % str(e))
             raise
 
+    def delete_listen(self, listened_at: int, user_name: str, recording_msid: str):
+        """ Delete a particular listen for user with specified MusicBrainz ID.
+
+        Args:
+            listened_at: The timestamp of the listen
+            user_name: the username of the user
+            recording_msid: the MessyBrainz ID of the recording
+
+        Raises: TimescaleListenStoreException if unable to delete the listen
+        """
+
+        args = {'listened_at': listened_at, 'user_name': user_name, 'recording_msid': recording_msid}
+        query = """DELETE FROM listen
+                    WHERE listened_at = :listened_at
+                      AND user_name = :user_name
+                      AND data -> 'track_metadata' -> 'additional_info' ->> 'recording_msid' = :recording_msid """
```

Comment:
Can we add an index for this? Otherwise it'll probably be very slow

Reply:
If possible, otherwise I'm happy to merge this.

Reply:
`CREATE INDEX recording_msid_ndx_listen ON listen (data -> 'track_metadata' -> 'additional_info' ->> 'recording_msid');`
Does this solve the issue

Reply:
As per our IRC discussion indexes are not needed.

Reply:
Upon some discussion and confusion on IRC it was noticed that index isn't much of use here as we already have an index on `user_name, listened_at`. So we think its not required as of now.